gordo_components/server/utils.py

```diff
@@ -224,20 +269,33 @@ def wrapper_method(self, *args, **kwargs):
 
         # Data provided by the client
         if request.method == "POST":
-            X = request.json.get("X")
-            y = request.json.get("y")
+
+            if request.json is not None:
+                X = request.json.get("X")
+                y = request.json.get("y")
+            else:
+                X = request.files.get("X")
+                y = request.files.get("y")
 
             if X is None:
                 message = dict(message='Cannot predict without "X"')
                 return make_response((jsonify(message), 400))
 
             # Convert X and (maybe) y into dataframes.
-            X = dataframe_from_dict(X)
+            if isinstance(X, FileStorage):
+                X = dataframe_from_parquet_bytes(X.read())
+            else:
+                X = dataframe_from_dict(X)
```

Comment:
This is exactly the same test as in line 273-278, but tested "opposite", and instead of using "request.json" it tests on the type if X. Maybe write it similar to the previous lines to make it easier to read?